packages/jsii-spec/lib/spec.ts

```diff
@@ -440,9 +449,9 @@ export interface Property extends Documentable, Overridable, SourceLocatable {
     name: string;
 
     /**
-     * The type of the property.
+     * The specification for the property's value (including it's declared type and whether it is optional).
      */
-    type: TypeReference;
+    value: TypeInstance<TypeReference>;
```

Comment:
Properties cannot be `promise`. They *can* be *optional* however.

Reply:
I can totes declare a `public readonly iSwearItWillHaveAValueAtSomePoint: Promise<string>`. If that's an API I *like*? No. But it can be made (and JSII will happily generate a `promise: true` for that type reference.

Reply:
See my previous comment, I'm not sure the kernel will handle that situation properly.